code/game/objects/items/devices/traitordevices.dm

```diff
@@ -276,16 +304,28 @@ effective or pretty fucking useless.
 	new /obj/item/wirecutters(src)
 
 /obj/item/storage/toolbox/emergency/turret/attackby(obj/item/attacking_item, mob/living/user, params)
-	if(attacking_item.tool_behaviour == TOOL_WRENCH && user.combat_mode && attacking_item.use_tool(src, user, 2 SECONDS, volume = 50))
-		user.visible_message(span_danger("[user] bashes [src] with [attacking_item]!"), \
-			span_danger("You bash [src] with [attacking_item]!"), null, COMBAT_MESSAGE_RANGE)
-		playsound(src, "sound/items/drill_use.ogg", 80, TRUE, -1)
-		var/obj/machinery/porta_turret/syndicate/toolbox/turret = new(get_turf(loc))
-		set_faction(turret, user)
-		turret.toolbox = src
-		forceMove(turret)
+	if(!istype(attacking_item, /obj/item/wrench/combat))
+		return ..()
+
+	if(!user.combat_mode)
+		return
+
+	if(!attacking_item.toolspeed)
 		return
-	return ..()
+
+	balloon_alert(user, "constructing...")
```

Comment:
these alerts should not be necessary as use tool should be putting up a progress bar

is that not happening?

Reply:
They are but i thought that it would look better with balloon alerts

Reply:
ok we disccused this in discord and it's fine

Reply:
Also i did not know other way to add balloon alert before start of do_after as it would appear after do_after finished